File: Prid1920-g03/Models/Vote.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;
using Microsoft.EntityFrameworkCore;

namespace Prid1920_g03.Models
{

    public class Vote
    {
        public int UpDown { get; set; }
        public int AuthorId { get; set; }
        public int PostId { get; set; }
        
        public virtual User User {get; set; }
        public virtual Post Post { get; set; }
        
    }
}

File: Prid1920-g03/ClientApp/src/app/models/vote.ts

export class Vote {
  updown: number; 
  authorId: number;
  postId: number;

  constructor(data: any) {
    this.updown = 1 || -1;
    this.authorId = data.authorId;
    this.postId = data.postId;
  }
}


